src/components/network/layers/arrow-layer.js

```diff
@@ -108,9 +108,17 @@ export default class ArrowLayer extends Layer {
 
     createTexture2D(gl, data, elementSize, format) {
         const elementCount = data.length / elementSize;
+        const width = Math.min(elementCount, this.state.maxTextureSize);
+        const height = Math.ceil(elementCount / this.state.maxTextureSize);
+        console.info(`Creating texture of ${elementCount} elements (${width} * ${height})`);
+        if (data.length < width * height * elementSize) {
+            const oldLength = data.length;
+            data.length += (width * height * elementSize - data.length);
```

Comment:
data.length = width * height * elementSize  ?

Reply:
Fixed, shame on me this is the ugliest line of code ever...